### PR TITLE
executor: support show create sequence

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve? 
Support `show create sequence` syntax.
After this PR, It goes like as follows:
```
mysql> create sequence seq;
Query OK, 0 rows affected (0.01 sec)

mysql> show create sequence seq;
+-------+----------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                               |
+-------+----------------------------------------------------------------------------------------------------------------------------+
| seq   | CREATE SEQUENCE `seq` start with 1 minvalue 1 maxvalue 9223372036854775806 increment by 1 cache 1000 nocycle ENGINE=InnoDB |
+-------+----------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
``` 

### What is changed and how it works?
1. add the `fetchShowCreateSequence` logic in `show.go`
2. append the `ast.ShowCreateSequence` case to the `ast.ShowCreateTable` case in `planbuilder.go` (in most case, sequence are treated as a table)
3. add some `show create sequence` test in sequence_test.go

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to update the documentation

Release note

 - Support the `show create sequence` in TiDB.

### CODE REVIEW

File: executor/show.go [hunk 1]
```diff
@@ -898,6 +900,40 @@ func ConstructResultOfShowCreateTable(ctx sessionctx.Context, tableInfo *model.T
...
+		fmt.Fprintf(buf, "cache %d ", sequenceInfo.CacheValue)
+	} else {
+		buf.WriteString("nocache ")
+	}
+	if sequenceInfo.Cycle {
+		buf.WriteString("cycle ")
+	} else {
+		buf.WriteString("nocycle ")
+	}
+	buf.WriteString("ENGINE=InnoDB")
```
Comment (crazycs520) [inline executor/show.go]:
```sql
test> create sequence seq;
Query OK, 0 rows affected
Time: 0.017s
test> show create sequence seq;
+-------+----------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                               |
+-------+----------------------------------------------------------------------------------------------------------------------------+
| seq   | CREATE SEQUENCE `seq` start with 1 minvalue 1 maxvalue 9223372036854775806 increment by 1 cache 1000 nocycle ENGINE=InnoDB |
+-------+----------------------------------------------------------------------------------------------------------------------------+
1 row in set
Time: 0.007s
test> drop sequence seq;
Query OK, 0 rows affected
Time: 0.014s
test> CREATE SEQUENCE `seq` start with 1 minvalue 1 maxvalue 9223372036854775806 increment by 1 cache 1000 nocycle ENGINE=InnoDB;
(8227, u'Unsupported sequence table-option InnoDB')

```

File: executor/show.go [hunk 2]
```diff
@@ -898,6 +900,40 @@ func ConstructResultOfShowCreateTable(ctx sessionctx.Context, tableInfo *model.T
 	return nil
 }
 
+// ConstructResultOfShowCreateSequence constructs the result for show create sequence.
+func ConstructResultOfShowCreateSequence(ctx sessionctx.Context, tableInfo *model.TableInfo, buf *bytes.Buffer) {
+	sqlMode := ctx.GetSessionVars().SQLMode
+	fmt.Fprintf(buf, "CREATE SEQUENCE %s ", escape(tableInfo.Name, sqlMode))
+	sequenceInfo := tableInfo.Sequence
```
Comment (crazycs520) [inline executor/show.go]:
```sql
test> create table t (a int key,b int);
Query OK, 0 rows affected
Time: 0.020s
test> show create sequence t;
(1105, u'runtime error: invalid memory address or nil pointer dereference')
```

File: ddl/sequence.go [hunk 3]
```diff
@@ -125,7 +125,7 @@ func handleSequenceOptions(SeqOptions []*ast.SequenceOption, sequenceInfo *model
 				sequenceInfo.Start = mathutil.MinInt64(sequenceInfo.MaxValue, model.DefaultNegativeSequenceStartValue)
 			}
 			if !minSetFlag {
-				sequenceInfo.MaxValue = model.DefaultNegativeSequenceMinValue
+				sequenceInfo.MinValue = model.DefaultNegativeSequenceMinValue
```
Comment (jackysp) [inline ddl/sequence.go]: bug fix?

File: executor/show.go (hunk 1 above)
Comment (AilinKid) [reply executor/show.go]: Got it.

File: executor/show.go (hunk 2 above)
Comment (AilinKid) [reply executor/show.go]: Got it.

File: ddl/sequence.go (hunk 3 above)
Comment (AilinKid) [reply ddl/sequence.go]: yes, A previous mistake, should open another PR to do this?

**Comment by AilinKid** [comment]
> BTW
> 
> ```sql
> test> show create sequence tt
> (1146, u"Table 'test.tt' doesn't exist")  -- shoud be Sequence xxx doesn't exist ?
> ```

It behavior like MariaDB... I think it's temporary ok here.

File: ddl/sequence.go (hunk 3 above)
Comment (jackysp) [reply ddl/sequence.go]: If the previous PR wasn't cherry-picked, I think just fix it in this PR is OK.

Comment (AilinKid) [reply ddl/sequence.go]: ok

File: ddl/sequence.go [hunk 4]
```diff
@@ -160,8 +160,6 @@ func buildSequenceInfo(stmt *ast.CreateSequenceStmt, ident ast.Ident) (*model.Se
 		switch op.Tp {
 		case ast.TableOptionComment:
 			sequenceInfo.Comment = op.StrValue
-		default:
```
Comment (XuHuaiyu) [inline ddl/sequence.go]: Why remove this line?

File: ddl/sequence.go [hunk 5]
```diff
@@ -160,6 +160,7 @@ func buildSequenceInfo(stmt *ast.CreateSequenceStmt, ident ast.Ident) (*model.Se
 		switch op.Tp {
 		case ast.TableOptionComment:
 			sequenceInfo.Comment = op.StrValue
+		case ast.TableOptionEngine:
```
Comment (XuHuaiyu) [inline ddl/sequence.go]: What does this mean?

Comment (AilinKid) [reply ddl/sequence.go]: Em... compatible with `create sequence seq Engine=InnoDB`, otherwise it will cast `Unsupported sequence table-option InnoDB`.

Comment (XuHuaiyu) [reply ddl/sequence.go]: It'll be better to add a comment for this

Comment (AilinKid) [reply ddl/sequence.go]: ok!

File: executor/show.go (hunk 1 above)
Comment (crazycs520) [reply executor/show.go]: Please add test for this.

Comment (crazycs520) [reply executor/show.go]: oh, i see it.

**Pull request merged**